internal/servers/controller/handlers/managed_groups/validate_test.go

```diff
@@ -0,0 +1,141 @@
+package managed_groups
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/boundary/internal/auth"
+	"github.com/hashicorp/boundary/internal/auth/oidc"
+	pb "github.com/hashicorp/boundary/internal/gen/controller/api/resources/managedgroups"
+	pbs "github.com/hashicorp/boundary/internal/gen/controller/api/services"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func fieldError(field, details string) string {
+	return fmt.Sprintf(`{name: %q, desc: %q}`, field, details)
+}
+
+func TestValidateCreateRequest(t *testing.T) {
+	cases := []struct {
+		name        string
+		item        *pb.ManagedGroup
+		errContains string
+	}{
+		{
+			name: "unrecognized authmethod prefix",
+			item: &pb.ManagedGroup{
+				AuthMethodId: "anything_1234567890",
+			},
+			errContains: fieldError(authMethodIdField, "Unknown auth method type from ID."),
+		},
+		{
+			name: "mismatched oidc authmethod pw type",
+			item: &pb.ManagedGroup{
+				Type:         "ldap",
```

Comment:
Oh how I miss the good old days of openLDAP.  Life and IAM was so much simpiler back then.